src/dmd/probing.d

```diff
@@ -0,0 +1,207 @@
+    static immutable const(char*) indent = "    ";
+    void print(string s)
+    {
+        if (!isFirst)
+            printf(",\n");
+        printf(`%s%s"%s"`, indent, indent, s.ptr);
+        isFirst = false;
+    }
+    fun(&print);
+    printf("\n%s]", indent);
+}
+
+private void determinePlatform(void delegate(string) print)
+{
+    import dmd.globals : global;
+
+    if (global.params.isWindows)
+    {
+	    print("windows");
+	}
+    else
+    {
+	    print("posix");
+        if (global.params.isLinux)
+	        print("linux");
+        if (global.params.isOSX)
+	        print("osx");
+        if (global.params.isFreeBSD)
+        {
+	        print("freebsd");
+	        print("bsd");
+	    }
+        if (global.params.isOpenBSD)
+        {
+	        print("openbsd");
+	        print("bsd");
+	    }
+        if (global.params.isSolaris)
+        {
+	        print("solaris");
+	        print("bsd");
+	    }
+    }
+    // Platforms not officially supported by DMD
+	version(NetBSD) print("netbsd");
+	version(DragonFlyBSD) print("dragonflybsd");
+	version(AIX) print("aix");
+	version(Haiku) print("haiku");
+	version(SkyOS) print("skyos");
+	version(SysV3) print("sysv3");
+	version(SysV4) print("sysv4");
+	version(Hurd) print("hurd");
+	version(Android) print("android");
+	version(Cygwin) print("cygwin");
+	version(MinGW) print("mingw");
+}
+
+private void determineArchitecture(void delegate(string) print)
+{
+    import dmd.globals : global;
+
+	if (global.params.is64bit)
+	{
+	    print("x86_64");
+	}
+	else
+	{
+	    version(X86) print("x86");
+	}
+
+    // Platforms not officially supported by DMD
+	version(ARM) print("arm");
+	version(ARM_Thumb) print("arm_thumb");
+	version(ARM_SoftFloat) print("arm_softfloat");
+	version(ARM_HardFloat) print("arm_hardfloat");
+	version(ARM64) print("arm64");
+	version(PPC) print("ppc");
+	version(PPC_SoftFP) print("ppc_softfp");
+	version(PPC_HardFP) print("ppc_hardfp");
+	version(PPC64) print("ppc64");
+	version(IA64) print("ia64");
+	version(MIPS) print("mips");
+	version(MIPS32) print("mips32");
+	version(MIPS64) print("mips64");
+	version(MIPS_O32) print("mips_o32");
+	version(MIPS_N32) print("mips_n32");
+	version(MIPS_O64) print("mips_o64");
+	version(MIPS_N64) print("mips_n64");
+	version(MIPS_EABI) print("mips_eabi");
+	version(MIPS_NoFloat) print("mips_nofloat");
+	version(MIPS_SoftFloat) print("mips_softfloat");
+	version(MIPS_HardFloat) print("mips_hardfloat");
+	version(SPARC) print("sparc");
+	version(SPARC_V8Plus) print("sparc_v8plus");
+	version(SPARC_SoftFP) print("sparc_softfp");
+	version(SPARC_HardFP) print("sparc_hardfp");
+	version(SPARC64) print("sparc64");
+	version(S390) print("s390");
+	version(S390X) print("s390x");
+	version(HPPA) print("hppa");
+	version(HPPA64) print("hppa64");
+	version(SH) print("sh");
+	version(SH64) print("sh64");
+	version(Alpha) print("alpha");
+	version(Alpha_SoftFP) print("alpha_softfp");
+	version(Alpha_HardFP) print("alpha_hardfp");
```

Comment:
Then leave it out ? No point in faking support, especially given those targets are more likely to be subject to cross compilation.